modules/theme-tools/social-menu.php

```diff
@@ -40,26 +40,61 @@ function jetpack_social_menu_init() {
 
 	// Enqueue CSS
 	add_action( 'wp_enqueue_scripts', 'jetpack_social_menu_style' );
+
+	// Load SVG icons related functions and filters
+	if ( 'svg' === jetpack_social_menu_get_type() ) {
+		require( dirname( __FILE__ ) . '/social-menu/icon-functions.php' );
+	}
 }
 add_action( 'after_setup_theme', 'jetpack_social_menu_init', 99 );
 
-/* Function to enqueue CSS */
+/**
+ * Return the type of menu the theme is using.
+ *
+ * @uses get_theme_support()
+ * @return $menu_type
+ */
+function jetpack_social_menu_get_type() {
+	$options = get_theme_support( 'jetpack-social-menu' );
+
+	if ( empty( $options ) ) {
+		$menu_type = null;
+	} else {
+		$menu_type = ( in_array( $options[0], array( 'genericons', 'svg' ) ) ) ? $options[0] : 'genericons';
+	}
+
+	return $menu_type;
+}
+
+/**
+ * Function to enqueue the CSS.
+ */
 function jetpack_social_menu_style() {
+	$menu_type = jetpack_social_menu_get_type();
+	$deps      = ( 'genericons' === $menu_type ) ? array( 'genericons' ) : null;
+
 	if ( has_nav_menu( 'jetpack-social-menu' ) ) {
-		wp_enqueue_style( 'jetpack-social-menu', plugins_url( 'social-menu/social-menu.css', __FILE__ ), array( 'genericons' ), '1.0' );
+		wp_enqueue_style( 'jetpack-social-menu', plugins_url( 'social-menu/social-menu-' . $menu_type . '.css', __FILE__ ), $deps, '1.0' );
```

Comment:
Should we check to make sure that `$menu_type` is a valid string (since jetpack_social_menu_get_type() can return null) before we enqueue it?